### PR TITLE
Added basic components and styling for listing demos

### PR DESCRIPTION
**Pull request by nimworks**
Added very basic styling for the **demo page** and the **demos list page**.
The styling would be updated later once the design language is finalized! 😇

### CODE REVIEW

File: src/pages/demo/custom-controls/index.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+- Cursor styling
+
+## Creating the control
+We create a new control and we add it to the object. To create a control we use `new fabric.Control(options)` In this case we want the corner to be near the top right classic control, so we give it a position of x of 0.5 that is the extreme right of the bounding box of the object. Same for y at -0.5. We do not want it to overlap with the current top right corner, so we add an `offsetY` with a positive number that will make it render a bit more down
+
+We want our control to render an icon rather than a fabric supported circle or square. So we have to provide our own render function to the control. The function is receiving the canvas context as first argument and so a simple drawImage will make it
+
+For the custom action we want simply to delete the object. We build a function following the mouseUpHandler signature and we use there `canvas.remove(target)`
+
+
```
Comment (nimworks) [inline src/pages/demo/custom-controls/index.md:27]: Here we can add the **CodePens** by embedding them in an iFrame as documented on [CodePen](https://blog.codepen.io/documentation/embedded-pens/)
Need to check the pricing caveats though, or maybe we can use offerings from other vendors.

**Comment by asturur** [comment]
You seems to have gigantic tabs. Would be possible to later to use the double space like in fabric.js?

File: src/pages/demo/custom-controls/index.md (hunk 1 above)
Comment (asturur) [reply src/pages/demo/custom-controls/index.md:27]: There is prefill embeds that is already working on the other website. So it will work here too.

**Comment by asturur** [comment]
I m sorry i can review just tomorrow! i m going in a very busy evening. If you have any screenshot of how it render locally, just paste it.

File: src/pages/demo/custom-controls/index.md (hunk 1 above)
Comment (nimworks) [reply src/pages/demo/custom-controls/index.md:27]: > You seems to have gigantic tabs. Would be possible to later to use the double space like in fabric.js?
Yes, I too noticed the TABS being gigantic! 😅

Sure, I cud switch to using double spaces, although I had been using TAB coz it helps configure the text editor to display tabs of desired width (either 2 or 4) when working on different screens!

Would check with a `.editorconfig` file id it helps! 😇

**Comment by nimworks** [comment]
> You seems to have gigantic tabs. Would be possible to later to use the double space like in fabric.js?

Yeah, I too noticed the tabs being too wide 😅
Perhaps they use 8 spaces in GitHub viewers by default

Sure, I cud switch to using double spaces, although I had been using TAB coz it helps configure the text editor to display tabs of desired width (either 2 or 4) when working on different screens!

Would check with a `.editorconfig` file if it helps! 😇

**Comment by nimworks** [comment]
FabricJS demos
**The demos listing page**
Its pretty raw atm, but would eventually reflect the design language once it's finalized along with the style-guide 🎨

**Pull request merged**